database/migration/migrate.go

```diff
@@ -0,0 +1,172 @@
+package migrations
+
+import (
+	"bytes"
+	"database/sql"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// LoadMigrations will load all migrations in the ./database/migration/migrations folder
+func (m *Migrator) LoadMigrations() error {
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == "migrationdir" {
+			MigrationDir = flag.Lookup("migrationdir").Value.String()
+		}
+	})
+
+	m.Log.Printf("Using migration folder %s\n", MigrationDir)
+
+	migration, err := filepath.Glob(MigrationDir + "/*.sql")
+
+	if err != nil {
+		return errors.New("failed to load migrations")
+	}
+
+	if len(migration) == 0 {
+		return errors.New("no migration files found")
+	}
+
+	sort.Strings(migration)
+
+	for x := range migration {
+		err = m.loadMigration(migration[x])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *Migrator) loadMigration(migration string) error {
+	file, err := os.Open(migration)
+	if err != nil {
+		return err
+	}
+
+	fileData := strings.Trim(file.Name(), MigrationDir)
+	fileSeq := strings.Split(fileData, "_")
+	seq, _ := strconv.Atoi(fileSeq[0])
+
+	b, err := ioutil.ReadAll(file)
+	if err != nil {
+		return err
+	}
+
+	up := bytes.Split(b, []byte("-- up"))
+
+	if len(up) == 1 {
+		return fmt.Errorf("invalid migration file %v", file.Name())
+	}
+
+	down := strings.Split(string(up[1]), "-- down")
+
+	temp := Migration{
+		Sequence: seq,
+		UpSQL:    down[0],
+		DownSQL:  down[1],
+	}
+
+	m.Migrations = append(m.Migrations, temp)
+
+	return nil
+}
+
+// RunMigration attempts to run current migrations against a database
+func (m *Migrator) RunMigration() (err error) {
+	v, err := m.getCurrentVersion()
+	if err != nil {
+		return
+	}
+	m.Log.Printf("Current database version: %v\n", v)
+
+	latestSeq := m.Migrations[len(m.Migrations)-1].Sequence
+
+	if v == latestSeq {
```

Comment:
I think we should also have a check+error for if the current database version is greater than the latestSeq as this would be considered alarming, even though it _should_ never happen

Reply:
Sounds good also just found a bug with latest version not being updated correct 

will get both pushed shortly